src/main.ts

```diff
@@ -73,18 +74,38 @@ class LocalTableland {
     // TODO: I don't think this is doing anything anymore...
     this.#_cleanup();
 
+    // check if the hardhat port is in use and retry 5 times (1 second between each)
+    const hardhatPort = 8545;
+    let retries = 0;
+    while (retries < 5) {
+      const portInUse = await checkPortInUse(hardhatPort);
+      if (!portInUse) break;
+
+      await new Promise((resolve) => setTimeout(resolve, 1000));
+      retries++;
+    }
+    // throw if all retries failed and port is in use
+    if (retries === 5) {
+      throw new Error(
+        `cannot start a local chain due to port ${hardhatPort} in use`
+      );
+    }
+
+    // You *must* store these in `process.env` to access within the hardhat subprocess
+    process.env.HARDHAT_NETWORK = "hardhat";
```

Comment:
here, i directly store the variables in process.env. if you don't do this and only pass them in the spawned child process, for some reason, they are never read (i.e., undefined within the hardhat scripts).